src/screens/Vault/VaultDetails.tsx

```diff
@@ -203,6 +156,34 @@ function VaultDetails({ navigation, route }: ScreenProps) {
   const exchangeRates = useExchangeRates();
   const currencyCode = useCurrencyCode();
   const currencyCodeExchangeRate = exchangeRates[currencyCode];
+  const { signerMap } = useSignerMap();
+  const { signers: vaultKeys } = vault || { signers: [] };
+
+  const { typeBasedIndicator } = useIndicatorHook({
+    types: [uaiType.SIGNING_DEVICES_HEALTH_CHECK],
+  });
+
+  const transformedArray = vaultKeys.map((item) => {
+    const signer: Signer = vaultKeys.length ? signerMap[item.masterFingerprint] : item;
+    const isRegistered = vaultKeys.length
+      ? item.registeredVaults.find((info) => info.vaultId === vault.id)
+      : false;
+    const isHealthCheckPending =
```

Comment:
We cant determine `isHealthCheckPending ` based on `typeBasedindiactor `. This might work now as it is derived from HC but if the dependency array changes this will become a garbage value. 

Let's change this logic to actual `lastHealthCheck` field value from th  actual signer.

Reply:
Lets make suer the data deriven UI logics are discussed with reps once to prevent rework. 
cc- @cakesoft-utkarsh

Reply:
` lastHealthCheck Threshold`   is 3 hours on dev and 90days on prod